### DISCUSSION TITLE
`node_link_graph` should have a keyword argument to use an arbitrary key for `nodes` in the dictionary

### DISCUSSION THREAD

**Discussion by bakhtos** [networkx/networkx#7531] [Ideas]
Currently, when loading graph data from a `JSON` (`dict`) with `node_link_graph`, it is possible to provide custom field names for `links`, `source`, `target` and `name`, but not for `nodes`, which means that it is necessary to first pre-process the dict before passing it on to a function.

**Comment by bakhtos** [comment]
Here is an example that caused me to post this:
https://github.com/M3SOulu/SARbenchmarks/blob/758e409224a53a91bb1975803a2b41b3e31871b2/graph_analysis.py
I am loading the data from this dataset https://tuhh-softsec.github.io/microSecEnD/anilallewar_microservices-basics-spring-boot.html

**Comment by dschult** [comment]
This seems like a reasonable request. I'm not sure why it was not made configurable already. I am guessing that the original code used only integer node values. A PR updated it so that non-integer nodes work, but the keyword name never got changed to being flexible.
